base/Base/Base_Base_implementation.F90

```diff
@@ -3169,6 +3169,174 @@ end function ijsearch
 
   end subroutine MAPL_GetHorzIJIndex
 
+  module subroutine MAPL_GetGlobalHorzIJIndex(npts,II,JJ,lon,lat,lonR8,latR8,Grid, rc)
+    implicit none
+    !ARGUMENTS:
+    integer,                      intent(in   ) :: npts ! number of points in lat and lon arrays
+    integer,                      intent(inout) :: II(npts) ! array of the first index for each lat and lon
+    integer,                      intent(inout) :: JJ(npts) ! array of the second index for each lat and lon
+    real, optional,               intent(in   ) :: lon(npts) ! array of longitudes in radians
+    real, optional,               intent(in   ) :: lat(npts) ! array of latitudes in radians
+    real(ESMF_KIND_R8), optional, intent(in   ) :: lonR8(npts) ! array of longitudes in radians
+    real(ESMF_KIND_R8), optional, intent(in   ) :: latR8(npts) ! array of latitudes in radians
+    type(ESMF_Grid),    optional, intent(inout) :: Grid ! ESMF grid
+    integer,            optional, intent(out  ) :: rc  ! return code
+
+    integer :: status
+    integer :: dims(3), IM_WORLD, JM_WORLD
+    real(ESMF_KIND_R8), allocatable, dimension (:,:) :: xyz
+    real(ESMF_KIND_R8), allocatable, dimension (:)   :: x,y,z
+    real(ESMF_KIND_R8), allocatable :: max_abs(:)
+    real(ESMF_KIND_R8)              :: sqr2, alpha, dalpha, error, shift, Japan_shift, c_lat
+    real(ESMF_KIND_R8), allocatable :: lons(:), lats(:)
+    real(ESMF_KIND_R8), allocatable :: corner_lons(:,:), corner_lats(:,:)
+    integer :: I1, I2, J1, J2, j
+    logical, save :: grid_is_ok = .false.
+
+
+    if (npts == 0 ) then
+      _RETURN(_SUCCESS)
+    endif
+
+    if ( .not. present(grid)) then
+      _ASSERT(.false., "need cubed-sphere grid")
+    endif
+    call MAPL_GridGet(grid, globalCellCountPerDim=dims,rc=status)
+    _VERIFY(STATUS)
+    IM_World = dims(1)
+    JM_World = dims(2)
+    _ASSERT( IM_WORLD*6 == JM_WORLD, "It only works for cubed-sphere grid")
+
+    ! define constants
+    sqr2   = sqrt(2.0d0) ! distance from center to the mid of an edge for a 2x2x2 cube
+    alpha  = asin(1.d0/sqrt(3.d0)) ! angle between the two lines(center to mid and center to end of an edge)
```

Comment:
I think these might already be available as MAPL constants.   If not, they should still be PARAMETERs in this procedure, rather than variables.

Reply:
now they become parameters. No calculations any more